### ISSUE TITLE
can't deserialize default(DateTime)

### ISSUE THREAD

**Issue by physhi** [rpgmaker/NetJSON#162]
The problem is that the format of the date in text is "1-01-01 00:00:00.0". The code assumes that the year portion will be 4 characters.

Test case NetJson.DeSerialize(NetJson.Serialize(default(DateTime))

**Comment by rpgmaker** [comment]
You need to use

NetJSONSetting.DateStringFormat = default: Null
             - This allows customization of date formatting and ignore other formatting option when specified.

It allows you to customize the serialization and deserialization of custom dates.

**Comment by physhi** [comment]
The problem is not really with just default(DateTime) but also with
default(DateTime).AddMinutes(1).

But I guess, I should use different dateTime serialization as they will be
much faster than string serializations (the main reason of using NetJson is
speed).

Anyways, great work, with the speed. Even though the code is a bit complex
(you are literally working with IL and it becomes hard to follow, but I
guess that's the only way to gurantee performance).

**Comment by rpgmaker** [comment]
Thanks 😀. Working with IL does not guaranteed speed. It is what it generate that does and the unsafe code implementation.

Do you think you can close this issue since you are using the standard date serialization?

Thanks,

**Comment by rpgmaker** [comment]
Closing this issue. Open if still needed
